api/liboni/drivers/test/onidriver_test.c

```diff
@@ -140,29 +142,34 @@ oni_driver_ctx oni_driver_create_ctx()
     // Start at zero and update in the loop below
     ctx->max_frame_size = 0;
 
-    int i;
-    for (i = 0; i < ctx->num_devs; i++) {
-
-        ctx->dev_table[i].dev.idx = i << 8; // All dev_idx 0 on different hubs
-        ctx->dev_table[i].dev.id = ONIX_TEST0;
-        ctx->dev_table[i].dev.version = 1;
-        ctx->dev_table[i].dev.read_size = 8 + 2 + 2 * (2 * i + 1); // [8: hub counter, 2: message word, 2 * (i + 1): dummy counter words]
-        ctx->dev_table[i].dev.write_size = 32;
-        ctx->dev_table[i].stream_enabled = 1;
-        ctx->dev_table[i].message = (uint16_t)(i * 42);
-        ctx->dev_table[i].dummy_words = 2 * i + 1; // This needs to be odd to enfornce 32-bit boundaries on frame data
-        ctx->dev_table[i].counter = 0;
-        ctx->dev_table[i].hubhwid = 5;
-        ctx->dev_table[i].hubfirmver = 1600;
-        ctx->dev_table[i].hubclkhz = (i + 1) * 50e6;
-        ctx->dev_table[i].hubdelayns = 628;
-
-        ctx->enabled_idx[i] = i;
-
-        ctx->max_frame_size
-            = ctx->max_frame_size < ctx->dev_table[i].dev.read_size ?
-                  ctx->dev_table[i].dev.read_size :
-                  ctx->max_frame_size;
+    assert(ctx->num_devs % NUMTESTDEVICESPERHUB == 0);  // Equal number of devices per hub
```

Comment:
Instead of a comment, provide an assertion failure message:
`assert(ctx->num_devs % NUMTESTDEVICESPERHUB == 0 && "Each hub must contain exactly NUMTESTDEVICESPERHUB devices");`

Reply:
I realize this is an assert, but it has a smell to it because this should be true by definition. The smell appears to be coming from something i wrote where n_devs is defined via dividing the device_table size and the bytes in a dev_t. Thats stupid. it should just be taken from the  NUMTESTDEVICESPERHUB  #define  since this is a completely static property. So in the end I would do that and remove the assert all together.